Distributions.ecl

```diff
@@ -0,0 +1,407 @@
+    REAL8 df_3 := df_2 * df;
+    REAL8 Term1 := B11*(d3+d1)/df;
+    REAL8 Term2 := B21*(B22*d7+B23*d5+B24*d3+B25*d1)/df_2;
+    REAL8 Term3 := B31*(B32*d11+B33*d9+B34*d7+B35*d5+B36*d3+B37*d1)/df_3;
+    REAL8 Norm_CDF := Normal_CDF(x);
+    T_CDF := MAP(df <= 0                             => 0.0,
+                df BETWEEN 3 AND 9     AND z<= -3000 => 0.0,
+                df BETWEEN 3 AND 9     AND z>=  3000 => 1.0,
+                df BETWEEN 10 AND 1000 AND z<= -150  => 0.0,
+                df BETWEEN 10 AND 1000 AND z>=  150  => 1.0,
+                df BETWEEN 1 AND 1000                => 0.5+sum_terms/2.0,
+                Norm_CDF-(EXP(-x*x*0.5))*(Term1+Term2+Term3)*DCONST);
+    RETURN T_CDF;
+  END;
+
+  // Inverse T helper functions
+  REAL8 df3_helper(REAL8 trm, REAL8 init_z) := BEGINC++
+    #option pure;
+    #include <math.h>
+    double c = cos(init_z);
+    double s = sin(init_z);
+    double z = init_z;
+    for (int i=0; i<15; i++) {
+      z = z-(z+s*c-trm)/(2.0*c*c);
+      c = cos(z);
+      s = sin(z);
+    }
+    return sqrt(3)*s/c;
+  ENDC++;
+  REAL8 df4_helper(REAL8 trm, REAL8 init_z) := BEGINC++
+    #option pure;
+    #include <math.h>
+    double c = cos(init_z);
+    double s = sin(init_z);
+    double z = init_z;
+    for (int i=0; i<15; i++) {
+      z = z-((1.0+0.5*c*c)*s-trm)/(1.5*c*c*c);
+      c = cos(z);
+      s = sin(z);
+    }
+    return sqrt(4)*s/c;
+  ENDC++;
+  REAL8 df5_helper(REAL8 trm, REAL8 init_z) := BEGINC++
+    #option pure;
+    #include <math.h>
+    double c = cos(init_z);
+    double s = sin(init_z);
+    double z = init_z;
+    for (int i=0; i<15; i++) {
+      z = z-(z+(c+(2.0/3.0)*c*c*c)*s-trm)/((8.0/3.0)*c*c*c*c);
+      c = cos(z);
+      s = sin(z);
+    }
+    return sqrt(5)*s/c;
+  ENDC++;
+  REAL8 df6_helper(REAL8 trm, REAL8 init_z) := BEGINC++
+    #option pure;
+    #include <math.h>
+    double c = cos(init_z);
+    double s = sin(init_z);
+    double z = init_z;
+    for (int i=0; i<15; i++) {
+      z = z-((1.0+0.5*c*c+0.375*c*c*c*c)*s-trm)/((15.0/8.0)*c*c*c*c*c);
+      c = cos(z);
+      s = sin(z);
+    }
+    return sqrt(6)*s/c;
+  ENDC++;
+  REAL8 dk_helper(REAL8 x, INTEGER4 df, REAL8 mf) := BEGINC++
+    #option pure;
+    #include<math.h>
+    const double last_position = 12; // lowest prob < 0.000001
+    const double step = last_position/100000;
+    double position = 0;
+    double cdf = 0.5;
+    double target = (x<0.5) ? 1.0-x :  x;
+    while (cdf < target && position <=last_position) {
+      double this_point = (2*position+step)/2;
+      cdf += mf*pow(1+(this_point*this_point/df), -0.5*(df+1))*step;
+      position += step;
+    }
+    if (cdf > target) position-= step/2;
+    position = (x<0.5)  ? -position  : position;
+    return position;
+  ENDC++;
+  /**
+   * Percentage point function for the T distribution.
+   * Translated from NIST SEL DATAPAC Fortran TPPF.f source
+   */
+  EXPORT REAL8 T_PPF(REAL8 x, REAL8 df) := FUNCTION
```

Comment:
You may want to add the DEFINE keyword here:
... := DEFINE FUNCTION
It would have the effect of generating an out of line function in the generated c++ so that if the function is used multiple times only one copy of the code will be generated.  Same goes for the T_CDF function above.